src/main/java/org/opentripplanner/transit/model/timetable/ScheduledTripTimes.java

```diff
@@ -53,18 +53,25 @@ public final class ScheduledTripTimes implements TripTimes {
   private final List<BookingInfo> dropOffBookingInfos;
   private final List<BookingInfo> pickupBookingInfos;
 
+  /**
+   * The headsign displayed by the vehicle at each stop along this trip. The array index is the
+   * stop index within this trip's TripPattern. Any number of array elements may point to the same
+   * I18NString instance if the headsign remains unchanged between stops.
+   */
   @Nullable
   private final I18NString[] headsigns;
 
   /**
-   * Implementation notes: This is 2D array since there can be more than one via name/stop per each
-   * record in stop sequence). Outer array may be null if there are no vias in stop sequence. Inner
-   * array may be null if there are no vias for particular stop. This is done in order to save
-   * space.
+   * A 2D array of String containing zero or more Via messages displayed at each stop in the
+   * stop sequence. This reference be null if no stop in the entire sequence of stops has any via
+   * strings. Any subarray may also be null or empty if no Via strings are displayed at that
+   * particular stop. These nulls are allowed to conserve memory in the common case where there are
+   * few or no via messages.
    */
   @Nullable
   private final String[][] headsignVias;
 
+  /** This is for matching stops with external services, in the realtime-updaters for example. */
```

Comment:
I have moved Javadoc up to interfaces and renamed the field to match interface accessor methods as requested. Even where information is pure pass-through I am still finding it a little awkward to remove all Javadoc, because although the IDE helps a bit, you still need to check all uses of the field, browse through the read uses to figure out which one is an accessor, go to that accessor, and then use something like cmd-U to find the right superclass (interface) method with the Javadoc. What do you think about using `@see` to show which interface method is reading through to the field, and allow the reader to find the Javadoc in one click (example in my latest commit)?